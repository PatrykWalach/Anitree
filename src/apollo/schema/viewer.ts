export interface User {
  id: number
  name: string
  avata: UserAvatar
  siteUrl: string
  options: {
    profileColor: string
    displayAdultContent: boolean
  }
  mediaListOptions: {
    scoreFormat: ScoreFormat
    animeList: MediaListTypeOptions
    mangaList: MediaListTypeOptions
  }
}
export interface UserAvatar {
  large: string
  medium: string
}
export type ScoreFormat =
  | 'POINT_100'
  | 'POINT_10_DECIMAL'
  | 'POINT_10'
  | 'POINT_5'
  | 'POINT_3'

export interface MediaListTypeOptions {
  customLists: string[]
  advancedScoring: string[]
  advancedScoringEnabled: boolean
}
