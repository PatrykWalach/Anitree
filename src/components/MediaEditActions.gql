
  fragment MediaEditActions_media on Media {
    id
    mediaListEntry {
      id
    }
  }

