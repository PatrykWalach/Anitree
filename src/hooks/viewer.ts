import { inject } from '@vue/composition-api'
import { useQuery } from '@vue/apollo-composable'
export const DefaultViewer = Symbol('Viewer')

export const useViewer = <V = ReturnType<typeof useQuery>>() =>
  inject(DefaultViewer) as V
