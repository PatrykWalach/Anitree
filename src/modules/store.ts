import * as modules from '.'
import { Store, plugin } from 'vuex-composition-api'

export const store =
  process.env.NODE_ENV === 'production'
    ? undefined
    : new Store({
        plugins: [plugin(Object.values(modules))],
      })
