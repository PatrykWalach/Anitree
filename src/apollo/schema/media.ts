import { ScoreFormat } from './viewer'
import { MediaList } from './mediaListCollection'

export interface Variables {
  isAdult?: boolean
  id: number
  format?: ScoreFormat
}

export interface Node {
  id: number
  type: MediaType
}

export type MediaType = 'MANGA' | 'ANIME'
export interface FuzzyDate {
  year: number | null
  month: number | null
  day: number | null
}

export interface Media extends Node {
  startDate: FuzzyDate
  endDate: FuzzyDate
  format: string | null
  description: string | null
  season: string | null
  seasonInt: number | null
  episodes: number | null
  chapters: number | null
  volumes: number | null
  isFavourite: boolean
  mediaListEntry: MediaList | null
  title: MediaTitle
  siteUrl: string
  relations: {
    edges: MediaEdge[]
  }
  tags: {
    rank: number
    name: string
  }[]
  studios: {
    nodes: Studio[]
  }
  status: MediaStatus | null
  bannerImage: string | null
  coverImage: MediaCoverImage
  stats: MediaStats
}
export interface MediaStats {
  scoreDistribution: ScoreDistribution[] | null
}
export interface ScoreDistribution {
  score: number
  amount: number
}
export interface MediaTitle {
  romaji: string
  english: string
  native: string
}

export type MediaRelation =
  | 'ADAPTATION'
  | 'PREQUEL'
  | 'SEQUEL'
  | 'PARENT'
  | 'SIDE_STORY'
  | 'CHARACTER'
  | 'SUMMARY'
  | 'ALTERNATIVE'
  | 'SPIN_OFF'
  | 'OTHER'
  | 'SOURCE'
  | 'COMPILATION'
  | 'CONTAINS'

export interface MediaEdge {
  node: Node
  relationType: MediaRelation
  id: number
}

export interface Studio {
  id: number
  name: string
}

export interface MediaCoverImage {
  extraLarge: string | null
  large: string | null
  medium: string | null
  color: string | null
}

export type MediaStatus =
  | 'CURRENT'
  | 'PLANNING'
  | 'COMPLETED'
  | 'DROPPED'
  | 'PAUSED'
  | 'REPEATING'
