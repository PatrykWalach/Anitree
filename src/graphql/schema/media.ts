import { ScoreFormat } from './viewer'
import { MediaList } from './mediaListCollection'

export interface Variables {
  isAdult?: boolean
  id: number
  format?: ScoreFormat
}

export interface Node {
  __typename: 'Media'
  id: number
  type: MediaType
}

export type MediaType = 'MANGA' | 'ANIME'
export interface FuzzyDate {
  __typename: 'FuzzyDate'
  year: number | null
  month: number | null
  day: number | null
}

export interface Media extends Node {
  startDate: FuzzyDate
  endDate: FuzzyDate
  format: string | null
  description: string | null
  season: string | null
  seasonInt: number | null
  episodes: number | null
  chapters: number | null
  volumes: number | null
  isFavourite: boolean
  ranknigs: MediaRank[]
  mediaListEntry: MediaList | null
  title: MediaTitle
  siteUrl: string
  relations: MediaConnection
  tags: MediaTag[]
  studios: StudioConnection
  status: MediaStatus | null
  bannerImage: string | null
  coverImage: MediaCoverImage
  stats: MediaStats
}
export interface MediaConnection {
  __typename: 'MediaConnection'
  edges: MediaEdge[]
}
export interface MediaTag {
  __typename: 'MediaTag'
  id: number
  rank: number
  name: string
}
export interface StudioConnection {
  __typename: 'StudioConnection'

  nodes: Studio[]
}
export interface MediaRank {
  __typename: 'MediaRank'
  allTime: boolean
  context: string
  rank: number
  type: 'RATED' | 'POPULAR'
  year: null | number
}
export interface MediaStats {
  __typename: 'MediaStats'
  scoreDistribution: ScoreDistribution[] | null
}
export interface ScoreDistribution {
  __typename: 'ScoreDistribution'
  score: number
  amount: number
}
export interface MediaTitle {
  __typename: 'MediaTitle'
  romaji: string
  english: string
  native: string
}

export type MediaRelation =
  | 'ADAPTATION'
  | 'PREQUEL'
  | 'SEQUEL'
  | 'PARENT'
  | 'SIDE_STORY'
  | 'CHARACTER'
  | 'SUMMARY'
  | 'ALTERNATIVE'
  | 'SPIN_OFF'
  | 'OTHER'
  | 'SOURCE'
  | 'COMPILATION'
  | 'CONTAINS'

export interface MediaEdge {
  __typename: 'MediaEdge'
  node: Node
  relationType: MediaRelation
  id: number
}

export interface Studio {
  __typename: 'Studio'
  id: number
  name: string
}

export interface MediaCoverImage {
  __typename: 'MediaCoverImage'
  extraLarge: string | null
  large: string | null
  medium: string | null
  color: string | null
}

export type MediaStatus =
  | 'CURRENT'
  | 'PLANNING'
  | 'COMPLETED'
  | 'DROPPED'
  | 'PAUSED'
  | 'REPEATING'
