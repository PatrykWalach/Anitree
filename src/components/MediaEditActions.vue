<template>
  <v-card-actions>
    <!-- <v-tooltip bottom>
      <template v-slot:activator="{ on, attrs }">
        <v-icon v-bind="attrs" v-on="on">
          {{ auto ? 'sync' : 'sync_disabled' }}
        </v-icon>
      </template>
      <span>{{ caption }}</span>
    </v-tooltip> -->

    <v-dialog v-model="confirmation" max-width="360">
      <template v-slot:activator="{ on, attrs }">
        <v-btn
          :disabled="!(user && media && media.mediaListEntry)"
          color="error"
          text
          v-bind="attrs"
          v-on="on"
          >Delete</v-btn
        >
      </template>
      <v-card>
        <v-card-title>Warning</v-card-title>
        <v-card-text
          >Are you sure you want to delete this list entry?</v-card-text
        >
        <v-card-actions>
          <v-spacer></v-spacer>
          <v-btn text color="error" @click="confirmation = false">Cancel</v-btn>

          <v-btn text color="primary" @click="remove">Ok</v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
    <v-spacer></v-spacer>

    <v-btn text color="primary" @click="close">Cancel</v-btn>

    <v-btn :disabled="!submitRequired" outlined color="primary" @click="submit"
      >Save</v-btn
    >
  </v-card-actions>
</template>
<script lang="ts">
import MediaEditIcon from './MediaEditIcon.vue'

import { Media } from '@/apollo/schema/media'
import { DELETE_MEDIA_LIST_ENTRY } from '@/apollo'
import {
  ref,
  computed,
  createComponent,
  SetupContext
} from '@vue/composition-api'
import { User } from '../apollo/schema/viewer'
import useEdit from '../store/edit'

export interface Props {
  user: User | null
  media: Media | null
}

function useActions(props: Readonly<Props>, { root }: SetupContext) {
  const confirmation = ref(false)

  const { close, form, submit: _submit, isEdited } = useEdit()

  const submitRequired = computed(() => !!Object.values(form.value).length)

  const submit = async () => {
    await _submit()

    isEdited.value = false
  }

  const remove = async () => {
    if (props.media && props.media.mediaListEntry) {
      confirmation.value = false
      await root.$apollo.mutate({
        mutation: DELETE_MEDIA_LIST_ENTRY,
        variables: { id: props.media.mediaListEntry.id }
      })
      isEdited.value = false
    }
  }

  return { confirmation, close, submitRequired, submit, remove }
}

export default createComponent<Readonly<Props>>({
  components: {
    MediaEditIcon
  },
  props: {
    user: { required: true, type: null, default: null },
    media: { required: true, type: null, default: null }
  },
  setup(props, context) {
    return useActions(props, context)
  }
})
</script>
