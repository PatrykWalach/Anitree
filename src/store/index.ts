import Vue from 'vue'
import Vuex from 'vuex'

// import { getModule } from 'vuex-module-decorators'
Vue.use(Vuex)

import { ModuleMedia } from './modules/media'
import { ModuleTheme } from './modules/theme'
import { ModuleTitle } from './modules/title'
import { ModuleAuth } from './modules/auth'

export interface RootState {
  media: ModuleMedia
  theme: ModuleTheme
  title: ModuleTitle
  auth: ModuleAuth
}

export default new Vuex.Store<RootState>({})
