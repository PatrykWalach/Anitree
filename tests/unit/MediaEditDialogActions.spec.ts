import MediaEditActions from '@/components/MediaEditActions.vue'
import { matchSnapshot } from './utils'
import { mockedMediaAndListEntry } from './mocks/media'
import { mockedViewer } from './mocks/viewer'
import { useMockedStore } from './mocks/store'

describe('MediaEditActions', () => {
  matchSnapshot(MediaEditActions, {
    propsData: {
      media: mockedMediaAndListEntry,
      viewer: mockedViewer,
    },
    provide: { ...useMockedStore() },
  })
})
