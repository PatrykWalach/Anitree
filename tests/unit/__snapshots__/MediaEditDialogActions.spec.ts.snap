// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaEditDialogActions matches snapshot 1`] = `
<v-card-actions-stub
  viewer="[object Object]"
>
  <v-dialog-stub
    closedelay="0"
    contentclass=""
    maxwidth="360"
    opendelay="0"
    origin="center center"
    retainfocus="true"
    transition="dialog-transition"
    width="auto"
  >
     
    <v-card-stub
      loaderheight="4"
      tag="div"
    >
      <v-card-title-stub>
        Warning
      </v-card-title-stub>
       
      <v-card-text-stub>
        Are you sure you want to delete this list entry?
      </v-card-text-stub>
       
      <v-card-actions-stub>
        <v-spacer-stub />
         
        <v-btn-stub
          activeclass=""
          color="error"
          tag="button"
          text="true"
          type="button"
        >
          Cancel
        </v-btn-stub>
         
        <v-btn-stub
          activeclass=""
          color="primary"
          tag="button"
          text="true"
          type="button"
        >
          Ok
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
   
  <v-spacer-stub />
   
  <v-btn-stub
    activeclass=""
    color="primary"
    outlined="true"
    tag="button"
    text="true"
    type="button"
  >
    Cancel
  </v-btn-stub>
   
  <v-btn-stub
    activeclass=""
    color="primary"
    depressed="true"
    disabled="true"
    tag="button"
    type="button"
  >
    Save
  </v-btn-stub>
</v-card-actions-stub>
`;
